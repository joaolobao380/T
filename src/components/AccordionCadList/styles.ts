import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#EBEBEB',
    padding: 16,
    borderRadius: 8,
  },
  containerTitleAndIcon: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  titleAccordion: {
    fontFamily: 'Poppins-Regular',
  },
});
