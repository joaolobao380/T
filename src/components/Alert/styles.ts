import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#BAE6FC',
    padding: 16,
    width: '100%',
    borderRadius: 8,
  },

  containerDescription: {
    flex: 1,
  },

  closeButton: {
    position: 'absolute',
    right: 16,
    top: 8,
  },

  containerIconAndChildren: {
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 20,
    gap: 8,
  },
});
