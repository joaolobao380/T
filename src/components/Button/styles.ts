import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  button: {
    paddingVertical: 16,
    backgroundColor: '#2A2A2A',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    borderRadius: 4,
  },
  loadingButton: {
    backgroundColor: '#444444',
  },
  pressedButton: {
    backgroundColor: '#1A1A1A',
  },
  buttonText: {
    color: '#FFFFFF',
    fontSize: 16,
    fontFamily: 'Poppins-Regular',
  },
});
