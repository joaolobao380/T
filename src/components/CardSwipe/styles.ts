import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  card: {
    backgroundColor: '#2A2A2A',
    borderRadius: 8,
    paddingVertical: 24,

    paddingHorizontal: 16,
    flexDirection: 'row',
    alignItems: 'center',
  },
  cardContent: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  title: {
    fontSize: 16,
    fontFamily: 'Poppins-Regular',
    color: 'white',
  },
  status: {
    fontSize: 14,
    color: 'gray',
  },
  tagContainer: {
    backgroundColor: '#6EFE85',
    borderRadius: 16,
    paddingHorizontal: 12,
    paddingVertical: 4,
  },
  tagContainerOff: {
    backgroundColor: '#DDDDDD',
    borderRadius: 16,
    paddingHorizontal: 12,
    paddingVertical: 4,
  },
  tag: {
    fontSize: 14,
    fontFamily: 'Poppins-Regular',
  },
  deleteButton: {
    backgroundColor: '#EC4848',
    justifyContent: 'center',
    alignItems: 'center',
    width: 80,
    height: '100%',
    marginLeft: -8,
    borderTopRightRadius: 8,
    borderBottomRightRadius: 8,
  },
  deleteButtonText: {
    color: 'white',
    fontWeight: 'bold',
  },
});
