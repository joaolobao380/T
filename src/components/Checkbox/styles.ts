import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
  },
  checkbox: {
    width: 24,
    height: 24,
    borderWidth: 1,
    borderColor: '#444444',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 2,
  },
  checked: {
    backgroundColor: '#444444',
  },
  label: {
    fontFamily: 'Poppins-Regular',
  },
});
