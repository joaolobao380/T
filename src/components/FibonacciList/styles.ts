import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  buttonSelected: {
    height: 40,
    width: 40,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#444444',
  },
  buttonNotSelected: {
    height: 40,
    width: 40,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#DBDBDB',
  },
  buttonTextSelected: {
    color: '#FFFFFF',
    fontSize: 14,
  },
  buttonTextNotSelected: {
    color: '#000000',
    fontSize: 14,
  },
  label: {
    color: '#2A2A2A',
    fontSize: 14,
    fontFamily: 'Poppins-Regular',
    marginBottom: 4,
  },
});
