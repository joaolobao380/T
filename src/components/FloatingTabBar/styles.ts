import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'transparent',
    borderRadius: 100,
    bottom: 20,
    marginHorizontal: 36,
    width: 167,
    height: 58,
    alignSelf: 'center',
    paddingHorizontal: 1,
    position: 'absolute',
  },
  tabButton: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  gradientBackground: {
    borderRadius: 100,
    height: '100%',
    width: '100%',
    flexDirection: 'row',
    paddingHorizontal: 1,
  },
  containerIconComponentFocused: {
    backgroundColor: 'white',
    padding: 15,
    borderRadius: 27,
    width: 54,
    height: 54,
    alignItems: 'center',
    justifyContent: 'center',
  },
  containerIconComponenteNotFocused: {
    backgroundColor: '#2A2A2A',
    padding: 0,
    borderRadius: 0,
    width: 0,
    height: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
