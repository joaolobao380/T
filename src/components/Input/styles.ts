import { StyleSheet } from 'react-native';
export const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  label: {
    color: '#2A2A2A',
    fontSize: 14,
    fontFamily: 'Poppins-Regular',
    marginBottom: 4,
  },
  input: {
    borderColor: '#2A2A2A',
    borderWidth: 1,
    backgroundColor: 'white',
    color: '#2A2A2A',
    fontSize: 16,
    fontFamily: 'Poppins-Regular',
    padding: 16,
    borderRadius: 5,
  },
  errorInput: {
    borderColor: 'red',
    borderWidth: 1,
  },
  errorText: {
    color: 'red',
    fontSize: 14,
    fontFamily: 'Poppins-Regular',
  },
});
