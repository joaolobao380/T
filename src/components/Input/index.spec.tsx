import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { useForm } from 'react-hook-form';

import { Input } from './index';
import { styles } from './styles';

function WrapperComponent({ errorMessage }: any) {
  const { control } = useForm({
    defaultValues: {
      testInput: '',
    },
  });

  return (
    <Input
      control={control}
      name="testInput"
      label="E-mail"
      placeholder="Ex: fulano@t.com"
      errorMessage={errorMessage}
    />
  );
}

describe('CustomInput', () => {
  it('renders correctly', () => {
    const { getByText } = render(<WrapperComponent errorMessage="" />);
    expect(getByText('E-mail')).toBeTruthy();
  });

  it('should accept text', () => {
    const { getByPlaceholderText } = render(<WrapperComponent errorMessage="" />);
    const input = getByPlaceholderText('Ex: fulano@t.com');
    fireEvent.changeText(input, 'test@test.com');
    expect(input.props.value).toEqual('test@test.com');
  });

  it('Should display error message when errorMessage prop is provided', () => {
    const errorMessage = 'E-mail é obrigatório';
    const { getByText } = render(<WrapperComponent errorMessage={errorMessage} />);
    expect(getByText(errorMessage)).toBeTruthy();
  });

  it('Should not display error message when errorMessage prop is not provided', () => {
    const { queryByText } = render(<WrapperComponent errorMessage="" />);
    expect(queryByText('E-mail é obrigatório')).toBeNull();
  });

  it('Should display red border and error message when errorMessage prop is provided', () => {
    const errorMessage = 'E-mail obrigatório';
    const { getByText, getByPlaceholderText } = render(
      <WrapperComponent errorMessage={errorMessage} />
    );
    expect(getByText(errorMessage)).toBeTruthy();
    const input = getByPlaceholderText('Ex: fulano@t.com');
    expect(input.props.style).toContainEqual(styles.errorInput);
  });

  it('renders a date picker when inputType is date', () => {
    jest.mock('@react-native-community/datetimepicker', () => 'DateTimePicker');

    const { getByTestId } = render(
      <WrapperComponent inputType="date" errorMessage="" defaultValue={new Date()} />
    );
    const datePicker = getByTestId('input_testID_normal');
    expect(datePicker).toBeTruthy();
  });

  it('renders correctly', () => {
    const { toJSON } = render(<WrapperComponent errorMessage="" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
