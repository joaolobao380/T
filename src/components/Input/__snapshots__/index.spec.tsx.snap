// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#2A2A2A",
        "fontFamily": "Poppins-Regular",
        "fontSize": 14,
        "marginBottom": 4,
      }
    }
  >
    E-mail
  </Text>
  <TextInput
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Ex: fulano@t.com"
    placeholderTextColor="#BBBBBB"
    secureTextEntry={false}
    style={
      [
        {
          "backgroundColor": "white",
          "borderColor": "#2A2A2A",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#2A2A2A",
          "fontFamily": "Poppins-Regular",
          "fontSize": 16,
          "padding": 16,
        },
        false,
      ]
    }
    value=""
  />
</View>
`;
